File: backend/src/main/java/com/example/taskmanagement/entity/Task.java
package com.example.taskmanagement.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Task implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "task_id")
    private Long taskId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id")
    private Project project;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_to")
    private Employee assignedTo;

    @Column(name = "title", nullable = false, length = 100)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "due_date")
    private LocalDate dueDate;

    @Column(name = "status", length = 50)
    private String status = "pending";

    @Column(name = "completed_at")
    private LocalDateTime completedAt;
}

File: mobile/lib/models/task.dart
class Task {
  final int taskId;
  final String title;
  final String? description;
  final String? dueDate;
  final String status;
  final int? projectId;
  final String? projectName;
  final int? assignedToId;
  final String? assignedToName;

  Task({
    required this.taskId,
    required this.title,
    this.description,
    this.dueDate,
    required this.status,
    this.projectId,
    this.projectName,
    this.assignedToId,
    this.assignedToName,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    final project = json['project'] as Map<String, dynamic>?;
    final assignedTo = json['assignedTo'] as Map<String, dynamic>?;
    return Task(
      taskId: json['taskId'] as int,
      title: json['title'] as String,
      description: json['description'] as String?,
      dueDate: json['dueDate'] as String?,
      status: json['status'] as String? ?? 'pending',
      projectId: project?['projectId'] as int?,
      projectName: project?['name'] as String?,
      assignedToId: assignedTo?['employeeId'] as int?,
      assignedToName: assignedTo != null
          ? '${assignedTo['firstName']} ${assignedTo['lastName']}'
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      if (description != null) 'description': description,
      if (dueDate != null) 'dueDate': dueDate,
      'status': status,
      if (projectId != null) 'projectId': projectId,
      if (assignedToId != null) 'assignedToId': assignedToId,
    };
  }
}
